azurerm/resource_arm_automation_certificate.go

```diff
@@ -0,0 +1,220 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationCertificate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationCertificateCreateUpdate,
+		Read:   resourceArmAutomationCertificateRead,
+		Update: resourceArmAutomationCertificateUpdate,
+		Delete: resourceArmAutomationCertificateDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"base64": {
```

Comment:
would a different name for this property make more sense? like `certificate_base64`?

Reply:
I did have a quick look at existing resources in the provider and aligned to those cases using `base64`. I figure in context of the resource declaration for `certificate` it's probably clear in context?

Reply:
Marking this as resolved as I think this is blocking having this looked at - and I think the argument that this aligns with existing `base64` certificate parameters is reasonable.

Definitely happy for more conversation around this however :)